### DISCUSSION TITLE
Button Click not compiling from xaml, but works in code behind

### DISCUSSION THREAD

**Discussion by sonicsmooth** [AvaloniaUI/Avalonia#10517] [General]
Preview 11.0.0-preview 4

Xaml button with `Click="OnClose"` or more directly `Click="Close"` yields "Unable to find setter..." and intellisense shows error:
OptionsDialogWindow.axaml.  Same with another event I tried `PointerEnter`.
![image](https://user-images.githubusercontent.com/672429/221956047-632e6818-9390-49ea-b9fa-5405b3ede321.png)

Same button from codebehind works fine.
OptionsDialogWindow.axaml.cs
![image](https://user-images.githubusercontent.com/672429/221956136-0627ad5b-5e85-4460-b4c0-c85dbf1bc524.png)

`Command=...` works fine.

thanks.

**Comment by timunie** [comment]
Where is OnClosed defined? I can't see it in your images posted.

**Reply by sonicsmooth** [reply]
I found my problem.  Whereas the lambdas that I could assign in the codebehind had `(o, e)` as arguments, the `OnClick` function referred to in the xaml did not declare `object o, RoutedEventArgs e` as arguments.
